### PR TITLE
Xdog changes

### PR DESCRIPTION
**Pull request by Afollet**

### CODE REVIEW

**Comment by Afollet** [comment]
Heavy refactor of Xdog. This allows for:

*Small classes with clearer behaviors
*Resuable code base -- shared with cbioportal transformer
*Increase testing coverage

Added Feature
*Harmonized/Unharmonized toggle
*Large omic data export

File: data-model/src/main/java/org/pdxfinder/graph/repositories/ModelCreationRepository.java [hunk 1]
```diff
@@ -208,19 +208,21 @@ Integer countMarkerAssociationBySourcePdxId(@Param("sourcePdxId") String sourceP
...
             "WHERE mc2.type = {type} "+
-            "WITH mod, iir, psamp, spr, sp, sfr, s, cbr, mc, mc2, cbr2, assoc, mAss, assoc2, mAss2 " +
+            "WITH iir, psamp, mod, samp, spr, sp, sfr, s, cbr, mc, mc2, cbr2, assoc, mAss, assoc2, mAss2 " +
             "OPTIONAL MATCH (sp)-[hsr:HOST_STRAIN]-(hs:HostStrain) " +
             "OPTIONAL MATCH (mc)-[pur:PLATFORM_USED]-(pl:Platform) " +
             "OPTIONAL MATCH (mc2)-[pur2:PLATFORM_USED]-(pl2:Platform) " +
-            "RETURN mod, iir, psamp, spr, sp, sfr, s, cbr, mc, mc2, cbr2, pur, pl, pur2, pl2, assoc, mAss, assoc2, mAss2, hsr, hs ")
-    ModelCreation findModelWithMolecularDataByDSAndIdAndMolcharType(@Param("dataSource") String dataSource, @Param("modelId") String modelId, @Param("type") String type);
+            "RETURN mod, iir, psamp, samp, spr, sp, sfr, s, cbr, mc, mc2, cbr2, pur, pl, pur2, pl2, assoc, mAss, assoc2, mAss2, hsr, hs ")
+    List<ModelCreation> findModelWithMolecularDataByDSAndIdAndMolcharType(@Param("dataSource") String dataSource, @Param("type") String type);
```
Comment (CsabaHalmagyi) [inline data-model/src/main/java/org/pdxfinder/graph/repositories/ModelCreationRepository.java:225]: Just a minor note: The id is not passed to this method so can be removed from the name, too.
findModelWithMolecularDataByDSAndMolcharType

File: data-model/src/main/java/org/pdxfinder/graph/repositories/ModelCreationRepository.java [hunk 2]
```diff
@@ -208,19 +208,21 @@ Integer countMarkerAssociationBySourcePdxId(@Param("sourcePdxId") String sourceP
     List<ModelCreation> findModelPlatformSampleByDS(@Param("dataSource") String dataSource);
 
 
-    @Query("MATCH (mod:ModelCreation) WHERE toLower(mod.dataSource) = toLower({dataSource}) and mod.sourcePdxId = {modelId} " +
+    @Query("MATCH (mod:ModelCreation) WHERE toLower(mod.dataSource) = toLower({dataSource}) " +
             "WITH mod " +
-            "OPTIONAL MATCH (mod)-[iir:IMPLANTED_IN]-(psamp:Sample)-[cbr:CHARACTERIZED_BY]-(mc:MolecularCharacterization)-[assoc:ASSOCIATED_WITH]->(mAss:MarkerAssociation) " +
+            "MATCH (mod)--(samp:Sample)-[cbr:CHARACTERIZED_BY]-(mc:MolecularCharacterization)-[assoc:ASSOCIATED_WITH]->(mAss:MarkerAssociation) " +
```
Comment (CsabaHalmagyi) [inline data-model/src/main/java/org/pdxfinder/graph/repositories/ModelCreationRepository.java:213]: If this is not an optional match, then only those samples will be returned here that have molecularcharacterization with data. Sometimes we have molchar nodes without markerassociation (so we can say that the model has molecular data just not in the finder). Disregard this if this is the intended behaviour.

Comment (CsabaHalmagyi) [reply data-model/src/main/java/org/pdxfinder/graph/repositories/ModelCreationRepository.java:213]: Also, (mod)--(samp:Sample) will return patient and xenograft samples as well.

Comment (Afollet) [reply data-model/src/main/java/org/pdxfinder/graph/repositories/ModelCreationRepository.java:213]: I want to grab all samples with marker associations so the data can be written into an omic data sheet. I want all the models with marker associations on either the patient sample or the models sample. However, I also want all the information necessary for the object mapper to fill out all the necessary data. I'm not sure what is necessary and what is not for the ModelCreation to get filled completely, but it seems to work.

**Pull request merged**